powertools-e2e-tests/src/test/java/software/amazon/lambda/powertools/ValidationE2ET.java

```diff
@@ -0,0 +1,99 @@
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import software.amazon.lambda.powertools.testutils.Infrastructure;
+import software.amazon.lambda.powertools.testutils.lambda.InvocationResult;
+
+class ValidationE2ET {
+
+	private static final ObjectMapper objectMapper = new ObjectMapper();
+
+	private static Infrastructure infrastructure;
+	private static String functionName;
+
+	@BeforeAll
+	@Timeout(value = 5, unit = TimeUnit.MINUTES)
+	public static void setup() {
+		infrastructure = Infrastructure.builder().testName(ValidationE2ET.class.getSimpleName())
+				.pathToFunction("validation").build();
+		Map<String, String> outputs = infrastructure.deploy();
+		functionName = outputs.get(FUNCTION_NAME_OUTPUT);
+	}
+
+	@AfterAll
+	public static void tearDown() {
+		if (infrastructure != null) {
+			infrastructure.destroy();
+		}
+	}
+
+	@ParameterizedTest
+	@Event(value = "/validation/valid_api_gw_in_out_event.json", type = APIGatewayProxyRequestEvent.class)
+	void test_validInboundApiGWEvent(APIGatewayProxyRequestEvent validEvent) throws IOException {
+		// WHEN
+		InvocationResult invocationResult = invokeFunction(functionName, objectMapper.writeValueAsString(validEvent));
+
+		// THEN
+		// invocation should pass validation and return 200
+		JsonNode validJsonNode = objectMapper.readTree(invocationResult.getResult());
+		assertThat(validJsonNode.get("statusCode").asInt()).isEqualTo(200);
+		assertThat(validJsonNode.get("body").asText()).isEqualTo("{\"price\": 150}");
+	}
+	
+	@ParameterizedTest
+	@Event(value = "/validation/invalid_api_gw_in_event.json", type = APIGatewayProxyRequestEvent.class)
+	void test_invalidInboundApiGWEvent(APIGatewayProxyRequestEvent validEvent) throws IOException {
+		// WHEN
+		InvocationResult invocationResult = invokeFunction(functionName, objectMapper.writeValueAsString(validEvent));
+
+		// THEN
+		// invocation should fail inbound validation and return 400
+		JsonNode validJsonNode = objectMapper.readTree(invocationResult.getResult());
+		assertThat(validJsonNode.get("statusCode").asInt()).isEqualTo(400);
+		assertThat(validJsonNode.get("body").asText()).contains("$.price: is missing but it is required");
+	}
+	
+	@ParameterizedTest
+	@Event(value = "/validation/invalid_api_gw_out_event.json", type = APIGatewayProxyRequestEvent.class)
+	void test_invalidOutboundApiGWEvent(APIGatewayProxyRequestEvent validEvent) throws IOException {
+		// WHEN
+		InvocationResult invocationResult = invokeFunction(functionName, objectMapper.writeValueAsString(validEvent));
+
+		// THEN
+		// invocation should fail outbound validation and return 400
+		JsonNode validJsonNode = objectMapper.readTree(invocationResult.getResult());
+		assertThat(validJsonNode.get("statusCode").asInt()).isEqualTo(400);
+		assertThat(validJsonNode.get("body").asText()).contains("$.price: must have an exclusive maximum value of 1000");
+	}
```

Comment:
could we add another test with something else than API GW, to make sure there is an exception thrown?

Reply:
The Function handler needs to be typed for the @Validation to works. Which means I need to add a second test function with something else than APIGW. I'm just not sure how to do it (in a compliant way) with the underlying Infrastructure builder that takes a single pathToFunction argument

Reply:
It will require a new function and a new E2ET.

Reply:
Added an E2E test for ALB events which demonstrate that in that case an exception is thrown